src/level-editor/levelEditorPage.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { ReactNode, useEffect, useRef } from 'react'
+import { useSelector, useDispatch } from '@dwalter/spider-hook'
+import {
+  tilesSelector,
+  addTile,
+  goatLocationSelector,
+  relocateGoat,
+  Location,
+  Tile as TileObject,
+  currentlyPlacingColorSelector,
+  cyclePlacingColor,
+  flipTile,
+} from './editorState'
+import { utils as SpiderUtils } from '@dwalter/spider-hook'
+
+const useEventListener = ((type: string, handler: Function, options: any) => {
+  const handlerRef = useRef(handler)
+
+  handlerRef.current = handler
+
+  const deps = [type, options]
+
+  const shouldUpdate = SpiderUtils.useShouldUpdate(deps)
+
+  useEffect(
+    shouldUpdate
+      ? () => {
+          const artificialHandler = (event: Event) => handlerRef.current(event)
+
+          addEventListener(type, artificialHandler, options)
+
+          return () => removeEventListener(type, artificialHandler, options)
+        }
+      : (null as any),
+    deps,
+  )
+}) as typeof addEventListener
+
+export function LevelEditorPage() {
+  const currentlyPlacingColor = useSelector(currentlyPlacingColorSelector)
+  const tiles = useSelector(tilesSelector)
+  const goatLocation = useSelector(goatLocationSelector)
+  const dispatch = useDispatch()
+
+  useEventListener('keydown', event => {
```

Comment:
Note the hook for dealing with the keyboard input so that we don't have those global variables